soh/soh/Enhancements/crowd-control/CrowdControl.cpp

```diff
@@ -142,8 +141,17 @@ void CrowdControl::ListenToServer() {
 
             // If effect is a one off run, let's execute
             if (!incomingEffect->timeRemaining) {
-                EffectResult result =
-                ExecuteEffect(incomingEffect->type.c_str(), incomingEffect->value, false);
+                EffectResult result = CrowdControl::CanApplyEffect(incomingEffect);
+                if (result == EffectResult::Success) {
+                    if (incomingEffect->category != "spawn_enemy") {
+                        GameInteractor::ApplyEffect(incomingEffect->giEffect);
+                    } else {
+                        if (!GameInteractor::SpawnEnemyWithOffset(incomingEffect->giEffect->value, incomingEffect->giEffect->value2)) {
+                            result = EffectResult::Retry;
+                        }
+                    }
+                    
+                }
```

Comment:
I think it's better like this -- then you no longer need to have that dummy effect for spawn.
```suggestion
                EffectResult result = EffectResult::Initiate;
                if (incomingEffect->category != "spawn_enemy") {
                      result = CrowdControl::CanApplyEffect(incomingEffect);
                      if (result == EffectResult::Success) {
                        GameInteractor::ApplyEffect(incomingEffect->giEffect);
                      }
                } else {
                      if (!GameInteractor::SpawnEnemyWithOffset(incomingEffect->giEffect->value, incomingEffect->giEffect->value2)) {
                            result = EffectResult::Retry;
                        }
                }
```

Reply:
The dummy effect is still there to check if a proper game is loaded and not paused. Moving that check into the SpawnEnemyWithOffset function kind of breaks the standard I was going for for all the other effects.

Reply:
I understand, but the fact you're calling it "dummy effect" already indicates it's abusing the system. You can just have two methods as part of GI:

```c++
static bool CanSpawnEnemy();
static uint8_t SpawnEnemyWithOffset(uint32_t enemyId, int32_t enemyParams);
```

then in CC you can change the above to:
```c++
EffectResult result = EffectResult::Retry;
if (incomingEffect->category == "spawn_enemy") {
    if (GameInteractor::Actions::CanSpawnEnemy() && GameInteractor::Actions::SpawnEnemyWithOffset(incomingEffect->giEffect->value, incomingEffect->giEffect->value2)) {
        result = EffectResult::Success;
    }
} else {
    result = CrowdControl::CanApplyEffect(incomingEffect);
    if (result == EffectResult::Success) {
        GameInteractor::ApplyEffect(incomingEffect->giEffect);
    }
}
```

It might've been nicer to just get the `EffectResult` result right away.. but it would've only been nice for using it in CC not necessarily for the API you're putting together here.